### PR TITLE
feat(metrics): export number of currently owned shards on node

### PR DESCRIPTION
**Pull request by mlegner**
Contributes to #1088.

### CODE REVIEW

File: crates/walrus-service/src/common/telemetry.rs [hunk 1]
```diff
@@ -416,21 +416,11 @@ pub(crate) struct CurrentEpochMetric(GenericGauge<AtomicU64>);
 
 impl CurrentEpochMetric {
     pub fn new() -> Self {
-        let opts = Opts::new("current_epoch", "The current Walrus epoch").namespace("walrus");
-        let metric = create_metric!(GenericGauge<AtomicU64>, opts, []);
-        Self(metric)
-    }
-
-    /// Registers the metric with the registry, and returns a new instance of it.
-    pub fn register(registry: &Registry) -> Self {
```
Comment (giac-mysten) [inline crates/walrus-service/src/common/telemetry.rs:425]: question: how come this is not needed anymore?

File: crates/walrus-service/src/common/telemetry.rs [hunk 2]
```diff
@@ -330,7 +330,7 @@ macro_rules! define_metric_set {
     (
         $(#[$outer:meta])*
         struct $name:ident {
-            $($new_type_field:ident: $new_type_field_type:ident),*
```
Comment (giac-mysten) [inline crates/walrus-service/src/common/telemetry.rs:333]: question: this was a bug, correct? the previous version allowed an arbitrary number of commas at the end?

File: crates/walrus-service/src/common/telemetry.rs (hunk 1 above)
Comment (mlegner) [reply crates/walrus-service/src/common/telemetry.rs:425]: I don't know, this isn't called anywhere and the `CurrentEpochStateMetric` also doesn't have it. I expect that the macros are now set up in a way that they automatically call the `register` function, but I'm not sure...

File: crates/walrus-service/src/common/telemetry.rs (hunk 2 above)
Comment (mlegner) [reply crates/walrus-service/src/common/telemetry.rs:333]: Not really a bug. The previous version didn't allow *any* comma at the end. The new version *requires* a comma at the end, which makes more sense when having the two types of fields. The combination would be possible with a `$(,)?`, but this doesn't seem to work as we have it already below and having it twice confuses the parser somehow.

**Pull request merged**